tools/autograd/derivatives.yaml

```diff
@@ -1,5 +1,46 @@
 # Defines derivative formulas and Python signatures of methods on Variable
 #
+# Each entry consists of:
+#   - A 'name', which specifies the ATen name of the function you
+#     are defining derivatives for, and a C-style argument
+#     specification.
+#   - One or more gradients entries, mapping a differentiable input
+#     names to a formula specifying how to compute its gradient.
+#     Note that a single gradient entry can specify the gradient
+#     formula for multiple input names, by specifying a key
+#     "input1, input2" (see atan2 for an example).
+#
+# Gradient expressions are standard C++ expressions operating on ATen
+# variables.  In a gradient expression, the following variables are in
+# scope:
+#   - 'grad' (aka 'grad_output'), the gradient of the output which
+#     we are going to left-multiply.  When the forward returns multiple
+#     outputs, 'grad' always refers to the first output
+#     (TODO: reconsider this); you can refer to other outputs using
```

Comment:
Yeah grads is fine. If you use `grad` we check that there is exactly one incoming gradient. (This is the common case). Otherwise, it's up to you to ensure that `grads` is the correct size.